docs/best-practices/css-scss.md

```diff
@@ -0,0 +1,140 @@
+---
+title: CSS/SCSS Guidelines
+---
+
+```
+Note: this document needs a refactor after we arrive to a new agreement
+for the Harmony Code Standard.
+```
+
+We aim (as far as possible) to…
+
+* think in terms of reusable visual patterns, not pages, e.g., button, dialog wrapper, menu item… it's similar to a component in Angular but doesn't always match 1:1
+* reduce the specificity of our CSS
+* use only CSS classes (no IDs, no elements)
+* keep our selector flat, no nesting
+
+That's why we use the [BEM naming convention](http://getbem.com/naming/) in our projects, please **read and understand the previous link**. As a quick reminder, BEM stands for `Block Element Modifier` which are the three components of the naming convention. The `Block` represents a _repeating visual pattern_ and it's meaningfull on it's own. The `Elements` are part of a block and has no standalone meaning. The `Modifier` are flags on `Blocks` or `Elements`, use them to change appearance, behavior or state. A summary of how its written:
+
+```css
+.block {}
+.block__element {}
+.block--modifier {}
+.block__element--modifier {}
+```
+
+For more in depth information see MJ University [CSS Methodologies working document](https://docs.google.com/document/d/14PQ_J4Ysh9VZALdE00B4hlcX_KO1HQVLrNkka7dMYbA/edit?usp=sharing).
+
+## How to organize the files in a SASS project
+
+```
+sass/
+├── main.scss
+├── _functions.scss
+├── _mixins.scss
+├── _vars.scss
+├── components/
+│   ├── _my-component.scss
+│   └── …
+└── elements/
+    ├── _html.scss
+    ├── _body.scss
+    └── …
+```
+
+* `main.scss` is the entry point, this file should only have `@import`s to other SASS files.
+* `_vars.scss` this holds all our global variables (if a variable is specific to a component it should go in the component file **inside** the top level class `.block {}`). As rule we name our variables from "general to specific", e.g.: preffer `$color-red`, or `$size-body` vs `$red-color` or `$body-size`.
```

Comment:
One point that we can discuss to add is to avoid too specific color var names. e.g.:
- used `$colo-primary` instead of `$color-red` 

So we can easily change and define themes

Reply:
Yep, that's better. This should be revisited (as it's 2 years old), we have a couple of taks in Asana to follow up on this:
- https://app.asana.com/0/1199525026761269/1199525026761284/f
- https://app.asana.com/0/1199525026761269/1199525026761316/f

Reply:
+1 can you add the change @Alex-DA ?

Reply:
Fixed in https://github.com/mobilejazz/harmony-reference/pull/82/commits/e0ae26b038e1f06bbe89881b6572a717cc1ec9ef